gantry/clients/prometheus/prometheus.py

```diff
@@ -0,0 +1,105 @@
+import logging
+import math
+import urllib.parse
+
+import aiohttp
+
+from gantry.clients.prometheus import util
+from gantry.clients.prometheus.job import PrometheusJobClient
+from gantry.clients.prometheus.node import PrometheusNodeClient
+
+
+class PrometheusClient:
+    def __init__(self, base_url: str, auth_cookie: str = ""):
+        # cookie will only be used if set
+        if auth_cookie:
+            self.cookies = {"_oauth2_proxy": auth_cookie}
+        else:
+            self.cookies = {}
+
+        self.base_url = base_url
+
+    async def query(self, type: str, **kwargs) -> dict:
```

Comment:
Can we break this function into `query` and `query_range`? I think that'll help simplify both code paths and allow us to take concrete parameters to both instead of kwargs.